zebra-chain/src/transparent.rs

```diff
@@ -132,6 +134,30 @@ impl Input {
             unreachable!("unexpected variant: Coinbase Inputs do not have OutPoints");
         }
     }
+
+    /// Get the value balance of this input.
+    ///
+    /// Needed to calculate the transparent value balance.
+    ///
+    /// # Panics
+    ///
+    /// If the provided Utxos don't have the transaction outpoint.
+    pub fn value_balance(
+        &self,
+        ordered_utxos: &HashMap<OutPoint, utxo::OrderedUtxo>,
+    ) -> Amount<NegativeAllowed> {
+        match self {
+            Input::PrevOut { outpoint, .. } => ordered_utxos
+                .get(outpoint)
+                .expect("Provided Utxos don't have transaction Outpoint")
+                .utxo
```

Comment:
Type Change:  We need to use `Utxo`s here, because that's what is stored in the state.

We also need to change the `Utxo` type in:
- [ ] the transaction and block `value_balance` methods
- [ ] the `remaining_transaction_value` method and function

Documentation/Method Name: This is a value or amount.

(The input and output value balances have opposite signs, but in this code, the values have the same sign. That's ok, we just need to document it.)

Can we change the method name as well?
`value` and `amount` are both accurate names.

```suggestion
    /// Get the value spent by this input.
    /// This amount is added to the transaction value pool by this input.
    ///
    /// # Panics
    ///
    /// If the provided Utxos don't have the transaction outpoint.
    pub fn value(
        &self,
        utxos: &HashMap<OutPoint, utxo::Utxo>,
    ) -> Amount<NegativeAllowed> {
        match self {
            Input::PrevOut { outpoint, .. } => utxos
                .get(outpoint)
                .expect("Provided Utxos don't have transaction Outpoint")
```